src/libs/sfdk/buildengine.cpp

```diff
@@ -497,6 +497,14 @@ void BuildEnginePrivate::updateVmProperties(const QObject *context, const Functo
     });
 }
 
+bool BuildEnginePrivate::isValid() const
+{
+    QTC_ASSERT(!sharedConfigPath.isEmpty(), return false);
+    QTC_ASSERT(!sharedSshPath.isEmpty(), return false);
+    QTC_ASSERT(virtualMachine->sshParameters().port(), return false);
+    return true;
```

Comment:
We should check all shared paths and other properties as well (if any).

Reply:
I guess we should. My original approach was to just check those properties which are needed for establishing the ssh connection, which should be enough to prevent the crash. Saying that the build engine or emulator is valid is a slightly different thing.